src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/orchestrator/network/_network_orchestrator.py

```diff
@@ -0,0 +1,37 @@
+from ...orchestrator._orchestrator import Orchestrator
+import ipaddress
+from avs.avsarconboarder.exception import InvalidInputError
+
+class NetworkOrchestrator(Orchestrator):
+    def __init__(self):
+        self.list_of_ip_addr_in_segment = list()
+
+    def validate_static_ip_cidr_block(self, config):
+        segmant_ip_cidr = config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"]
+        gateway_ip      = ""
+        char_index      = 0
+        while char_index < len(segmant_ip_cidr) and segmant_ip_cidr[char_index] != '/':
+            gateway_ip += segmant_ip_cidr[char_index]
+            char_index += 1
+
+        cidr = segmant_ip_cidr[char_index - len(segmant_ip_cidr) + 1:]
+        if int(cidr) != 28:
+            raise InvalidInputError("Invalid segamnt block size provided, Please provide a /28 address")
+        
+        try:
+            subnet_first_ip_addr = ipaddress.IPv4Address(gateway_ip) - 1
+            self.list_of_ip_addr_in_segment = list(ipaddress.ip_network('{}/{}'.format(subnet_first_ip_addr,cidr)).hosts())
+        except:
+            raise InvalidInputError("Invalid gateway ip provided, please provide a segmant with first gateway ip")          
+    
+    def populate_network_config(self, config):
```

Comment:
>populate_network_config [](http://example.com/codeflow?start=8&length=23)

Can we add the IP distribution we are following indicating the free pool and reserved ips in a comment block here? Will help build context for future changes

Reply:
sure

Reply:
commented it before the function populate_network_config